planning/behavior_path_planner/autoware_behavior_path_goal_planner_module/src/goal_planner_module.cpp

```diff
@@ -1153,20 +1289,24 @@ std::optional<PullOverPath> GoalPlannerModule::selectPullOverPath(
   // STEP1: Filter valid paths before sorting
   // NOTE: Since copying pull over path takes time, it is handled by indices
   std::vector<size_t> sorted_path_indices;
-  sorted_path_indices.reserve(pull_over_path_candidates.size());
+  if (!sorted_bezier_indices_opt) {
+    sorted_path_indices.reserve(pull_over_path_candidates.size());
 
-  // STEP1-1: Extract paths which have safe goal
-  // Create a map of goal_id to GoalCandidate for quick access
-  std::unordered_map<int, GoalCandidate> goal_candidate_map;
-  for (const auto & goal_candidate : goal_candidates) {
-    goal_candidate_map[goal_candidate.id] = goal_candidate;
-  }
-  for (size_t i = 0; i < pull_over_path_candidates.size(); ++i) {
-    const auto & path = pull_over_path_candidates[i];
-    const auto goal_candidate_it = goal_candidate_map.find(path.goal_id());
-    if (goal_candidate_it != goal_candidate_map.end() && goal_candidate_it->second.is_safe) {
-      sorted_path_indices.push_back(i);
+    // STEP1-1: Extract paths which have safe goal
+    // Create a map of goal_id to GoalCandidate for quick access
+    std::unordered_map<int, GoalCandidate> goal_candidate_map;
+    for (const auto & goal_candidate : goal_candidates) {
+      goal_candidate_map[goal_candidate.id] = goal_candidate;
+    }
+    for (size_t i = 0; i < pull_over_path_candidates.size(); ++i) {
+      const auto & path = pull_over_path_candidates[i];
+      const auto goal_candidate_it = goal_candidate_map.find(path.goal_id());
+      if (goal_candidate_it != goal_candidate_map.end() && goal_candidate_it->second.is_safe) {
+        sorted_path_indices.push_back(i);
+      }
     }
+  } else {
+    sorted_path_indices = sorted_bezier_indices_opt.value();
```

Comment:
why skip goal safety check?

Reply:
fixed in bc8f304f9598ed381c9920c56d0da82fc82ac63a